### ISSUE TITLE
Failed unit : rpm-ostree-fix-shadow-mode.service

### ISSUE THREAD

**Issue by Nemric** [coreos/fedora-coreos-tracker#1722]
### Describe the bug

Booting a next stream FCOS 40 I can see when login to node :
```
[systemd]
Failed Units: 1
  rpm-ostree-fix-shadow-mode.service
```
I'm using a diskless PXE provisionned machine 

I have this logs
```
journalctl -eu rpm-ostree-fix-shadow-mode.service
Apr 30 17:30:45 Turing systemd[1]: Starting rpm-ostree-fix-shadow-mode.service - Update permissions for /etc/shadow...
Apr 30 17:30:45 Turing rpm-ostree[1250]: error: Read-only file system
Apr 30 17:30:45 Turing systemd[1]: rpm-ostree-fix-shadow-mode.service: Main process exited, code=exited, status=1/FAILURE
Apr 30 17:30:45 Turing systemd[1]: rpm-ostree-fix-shadow-mode.service: Failed with result 'exit-code'.
Apr 30 17:30:45 Turing systemd[1]: Failed to start rpm-ostree-fix-shadow-mode.service - Update permissions for /etc/shadow.
```

### Reproduction steps

Boot from PXE with http ign provisionning

### Expected behavior

No failed units

### Actual behavior

Failed Unit

### System details

Bare metal PXE booted and provisonned node

**Comment by Nemric** [comment]
I'm using this https://docs.fedoraproject.org/en-US/fedora-coreos/live-booting/#_booting_via_pxe
A mix between pxe and ipxe, but that works fine
Do we need this service to run as it seems to be here for updating an installed FCOS ?
I think the shadow mode is correct on a stock pxe booted node isn't it ?

**Comment by jlebon** [comment]
Yeah, I think we basically just need a `ConditionPathExists=!/run/ostree-live` in that unit.

Or... maybe `ConditionKernelCommandLine=ostree` would be more accurate. `/run/ostree-live`, while it sounds like an ostree thing, is actually written by CoreOS code. Whereas `ConditionKernelCommandLine=ostree` is pretty canon and a more generic fix.

**Comment by dustymabe** [comment]
Upstream fix in https://github.com/coreos/rpm-ostree/pull/4944

**Comment by Nemric** [comment]
Did this PR was merged into current (40.20240519.1.0) "next stream" ? I still see this failed service :/

**Comment by jlebon** [comment]
> Did this PR was merged into current (40.20240519.1.0) "next stream" ? I still see this failed service :/

It just made it into a new rpm-ostree release. It should show up in the next releases of `next` and `testing`.

**Comment by Nemric** [comment]
Ok, works great on 40.20240616.1.0 (mid june next stream)
Is there something else to check before closing this issue ? Like ...  is "rpm-ostree-shadow-mode" fixed ^^

**Comment by jbtrystram** [comment]
You can look that the permissions on /etc/shadow are correct (600)
On live systems it doesn't matter that's why we disabled it.

**Comment by travier** [comment]
This landed in [rpm-ostree 2024.6](https://github.com/coreos/rpm-ostree/releases/tag/v2024.6) which landed in Fedora CoreOS next in 40.20240616.1.0, testing in 40.20240616.2.0. It's not yet in stable.

**Comment by jbtrystram** [comment]
I booted a liveISO of 40.20240616.3.0 yesterday and did not see the message
